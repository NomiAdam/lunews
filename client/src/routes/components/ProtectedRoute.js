import React, { useContext } from 'react';
import { Route } from 'react-router-dom';
import { isNilOrEmptyString } from 'ramda-extension';
import { UserContext } from '../../context/userContext';
import Login from '../../pages/Login';

const ProtectedRoute = ({ component: Component, ...other }) => {
    const { user: { profile } } = useContext(UserContext);
    return (
        <Route
            { ...other }
            render={
                props => (isNilOrEmptyString(profile) ? (
                    <Login />
                ) : (
                    <Component { ...props } />
                ))
            }
        />
    );
};

export default ProtectedRoute;
