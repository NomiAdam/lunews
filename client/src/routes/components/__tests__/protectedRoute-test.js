/* eslint-disable no-undef */
import React from 'react';
import { shallow } from 'enzyme';
import ProtectedRoute from '../ProtectedRoute';

describe('ProtectedRoute', () => {
    it('should be defined', () => {
        expect(ProtectedRoute).toBeDefined();
    });
    it('should match the snapshot', () => {
        expect(shallow(<ProtectedRoute />)).toMatchSnapshot();
    });
});
