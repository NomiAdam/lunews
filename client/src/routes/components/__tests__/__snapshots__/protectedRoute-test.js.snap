// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProtectedRoute should match the snapshot 1`] = `<[object Object] />`;
